helix-term/src/ui/editor.rs

```diff
@@ -441,11 +441,18 @@ impl EditorView {
             // extra loops if the code is deeply nested.
 
             for i in starting_indent..(indent_level / tab_width as u16) {
+                let style = if config.indent_guides.rainbow {
+                    let color_index = i as usize % theme.rainbow_length();
+                    indent_guide_style.patch(theme.get(&format!("rainbow.{}", color_index)))
```

Comment:
I think we can speed up the calculation of the rainbow color: `Theme` can have a new associated function that returns the style for the given usize

```rust
pub fn get_rainbow(&self, index: usize) -> Style {
    self.highlights[index % self.rainbow_length]
}
```

```rust
indent_guide_style.patch(theme.get_rainbow(i));
```

Since rainbow highlights are the first `self.rainbow_length` elements of the `self.highlights` Vec

Reply:
> I think we can speed up the calculation of the rainbow color: `Theme` can have a new associated function that returns the style for the given usize
> 
> ```rust
> pub fn get_rainbow(&self, index: usize) -> Style {
>     self.highlights[index % self.rainbow_length]
> }
> ```
> 
> ```rust
> indent_guide_style.patch(theme.get_rainbow(i));
> ```
> 
> Since rainbow highlights are the first `self.rainbow_length` elements of the `self.highlights` Vec

This serves a lot of benefits.
1. no string comparisons
2. better readability
3. you may don't have to read the code and can rely on the functions name if you want to use a rainbow